### PR TITLE
Fixed unchecked None value in SubprocVecEnv

### PR DESCRIPTION
**Pull request by ScheiklP**
## Description
Fix for bug described in https://github.com/DLR-RM/stable-baselines3/issues/807

closes https://github.com/DLR-RM/stable-baselines3/issues/807

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:
- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

### CODE REVIEW

File: stable_baselines3/common/vec_env/subproc_vec_env.py [hunk 1]
```diff
@@ -123,6 +123,8 @@ def step_wait(self) -> VecEnvStepReturn:
         return _flatten_obs(obs, self.observation_space), np.stack(rews), np.stack(dones), infos
 
     def seed(self, seed: Optional[int] = None) -> List[Union[None, int]]:
+        if seed is None:
```
[the remark this answers is not in the document]
Comment (ScheiklP) [reply stable_baselines3/common/vec_env/subproc_vec_env.py:126]: Ah, no, I just did not check. Fixed in a new commit.

File: stable_baselines3/common/vec_env/dummy_vec_env.py [hunk 2]
```diff
@@ -51,6 +51,8 @@ def step_wait(self) -> VecEnvStepReturn:
         return (self._obs_from_buf(), np.copy(self.buf_rews), np.copy(self.buf_dones), deepcopy(self.buf_infos))
 
     def seed(self, seed: Optional[int] = None) -> List[Union[None, int]]:
+        if seed is None:
```
Comment (araffin) [inline stable_baselines3/common/vec_env/dummy_vec_env.py:54]: why not just return in that case?

Comment (ScheiklP) [reply stable_baselines3/common/vec_env/dummy_vec_env.py:54]: From intuition I would assume that if I call `env.seed()` without parameters, that a seed would be chosen for me randomly.
I mostly want to ensure, that all envs of the vecenv have a different seed. If we just return, all of them will have the same seed, when the processes are forked.

Comment (araffin) [reply stable_baselines3/common/vec_env/dummy_vec_env.py:54]: I see your point, I thought that the seed would have been different per process but a quick test shows it's not the case when processes are forked (which is not the default method of SB3 luckily), so the following yield a different result per process:

```python
import multiprocessing as mp

import numpy as np

def f(x):
    return np.random.rand()

if __name__ == '__main__':
    ctx = mp.get_context('spawn')
    with ctx.Pool(3) as p:
        print(p.map(f, [1, 1, 1]))
```

Comment (ScheiklP) [reply stable_baselines3/common/vec_env/dummy_vec_env.py:54]: Huh, for me the output of your example is the same over all processes with fork.
```
import multiprocessing as mp

import numpy as np

def f(x):
    return np.random.rand()

if __name__ == '__main__':
    ctx = mp.get_context('fork')
    with ctx.Pool(3) as p:
        print(p.map(f, [1, 1, 1]))
```
prints
```
[0.5667669684783624, 0.5667669684783624, 0.5667669684783624]
```

Yeah, sure, the `seed = np.random.randint(0, 2**32 - 1)` will be deterministic, if you force a seed before, but the subprocesses will still have their own unique seed.

Comment (araffin) [reply stable_baselines3/common/vec_env/dummy_vec_env.py:54]: > Huh, for me the output of your example is the same over all processes with fork.

yes, but that's not the case for `SuprocVecEnv` with start method fork for some reasons... (see test below, you can comment out `seed = np.random.randint(0, 2**32 - 1)` part and return None to check what happens)

Comment (araffin) [reply stable_baselines3/common/vec_env/dummy_vec_env.py:54]: > with start method fork for some reasons...

nevermind, I think the example in the test is using a separate random seed generator. Let me try.

Comment (araffin) [reply stable_baselines3/common/vec_env/dummy_vec_env.py:54]: See https://github.com/DLR-RM/stable-baselines3/pull/808/commits/a64714b546b82841bed456c53c36c6e84fec6392

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**